Sources/TuistGraph/Models/CompatibleXcodeVersions.swift

```diff
@@ -1,20 +1,70 @@
 import Foundation
+import struct ProjectDescription.Version
 
 /// Enum that represents all the Xcode versions that a project or set of projects is compatible with.
-public enum CompatibleXcodeVersions: Equatable, Hashable, ExpressibleByArrayLiteral {
+public enum CompatibleXcodeVersions: Equatable, Hashable, ExpressibleByArrayLiteral, ExpressibleByStringInterpolation,
+    CustomStringConvertible
+{
     /// The project supports all Xcode versions.
     case all
 
+    case exact(Version)
+
+    case upToNextMajor(Version)
+
+    case upToNextMinor(Version)
+
     /// List of versions that are supported by the project.
-    case list([String])
+    case list([CompatibleXcodeVersions])
+
+    public func isCompatible(versionString: String) -> Bool {
+        let xCodeVersion: Version = "\(versionString)"
+
+        switch self {
+        case .all:
+            return true
+        case let .exact(version):
+            return version == xCodeVersion
+        case let .upToNextMajor(version):
+            return version == xCodeVersion || (version.major == xCodeVersion.major && xCodeVersion > version)
+        case let .upToNextMinor(version):
+            return version == xCodeVersion ||
+                (version.major == xCodeVersion.major && version.minor == xCodeVersion.minor && xCodeVersion > version)
+        case let .list(versions):
+            return versions.map { $0.isCompatible(versionString: versionString) }.contains(true)
+        }
+    }
+
+    // MARK: - ExpressibleByStringInterpolation
+
+    public init(stringLiteral value: String) {
+        self = .exact("\(value)")
```

Comment:
I find this less surprising, same for the others below
```suggestion
        self = .exact(Version(stringLiteral: value))
```